lib/ProductOpener/DataQualityFood.pm

```diff
@@ -816,14 +816,27 @@ sub check_nutrition_data ($product_ref) {
 				$nutriments_values_occurences{$nutriment_value} = 1;
 			}
 		}
-		# raise warning if there are 3 or more duplicates in nutriments
+		# raise warning if there are 3 or more duplicates in nutriments and nutriment is above 1
 		foreach my $keys (keys %nutriments_values_occurences) {
-			if ($nutriments_values_occurences{$keys} > 2) {
+			if (($nutriments_values_occurences{$keys} > 2) and ($keys > 1)) {
 				push @{$product_ref->{data_quality_warnings_tags}}, "en:nutrition-3-or-more-values-are-identical";
 				last;
 			}
 		}
 
+		# retrieve max number of occurences
```

Comment:
I think that you can do something simpler that would also work when people put a non zero value (e.g. 1) in all fields:

Just check if you have only 1 key, or you have 2 keys and 1 of them is "salt_100g" or "sodium_100g". (and also that you have at least 3 or 4 entries in @major_nutriments_values)

Reply:
I did something like that. But, because it is an error, I would not add that part "and also that you have at least 3 or 4 entries" for now.

Note that there are not much more rows if we count salt or sodium instead of both (about 10 more): [mirabelle](https://mirabelle.openfoodfacts.org/products?sql=select+*+from+%5Ball%5D+where+%22energy-kj_100g%22+%3D+%22energy-kcal_100g%22+and+%22energy-kcal_100g%22+%3D+%22fat_100g%22+and+%22fat_100g%22+%3D+%22saturated-fat_100g%22+and+%22saturated-fat_100g%22+%3D+%22carbohydrates_100g%22+and+%22carbohydrates_100g%22+%3D+%22sugars_100g%22+and+%22sugars_100g%22+%3D+%22fiber_100g%22+and+%22fiber_100g%22+%3D+%22proteins_100g%22+and+%28%22proteins_100g%22+%3D+%22salt_100g%22+or+%22proteins_100g%22+%3D+%22sodium_100g%22%29%0D%0Aorder+by+random%28%29+limit+10)